### PR TITLE
Rename suspend-to-hibernate to suspend-then-hibernate

### PR DESCRIPTION
**Pull request by superm1**
Per some discussion with @hadess on the Gnome pieces, they would prefer this name
as it's more descriptive of what's happening. @keszybz mentioned in #8274 they were amenable to renaming.

### CODE REVIEW

File: units/suspend-then-hibernate.target [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Suspend; Idle into hibernate
```
Comment (keszybz) [inline units/suspend-then-hibernate.target]: That was already before this PR, I just noticed it: this Description does not make sense — "Idle into hibernate"?

Something wrong happens in the rename:
before:
units/suspend-to-hibernate.target
units/systemd-suspend-to-hibernate.service.in
after:
units/suspend-then-hibernate.service
units/suspend-then-hibernate.target
units/systemd-suspend-then-hibernate.service.in

Comment (keszybz) [reply units/suspend-then-hibernate.target]: (Please make the change of the Description a separate commit.)

Comment (superm1) [reply units/suspend-then-hibernate.target]: Thanks for catching that, I thought something seemed off and was confused why git wasn't renaming one of the files.

I've fixed both of these.

File: units/suspend-then-hibernate.target [hunk 2]
```diff
@@ -8,7 +8,7 @@
 #  (at your option) any later version.
 
 [Unit]
-Description=Suspend; Idle into hibernate
+Description=Suspend; Hibernate if not used for a period of time.
```
Comment (keszybz) [inline units/suspend-then-hibernate.target]: Please remove the periods from the end (Description is used as a phrase in sentences, so we'd have double punctuation.)

Comment (superm1) [reply units/suspend-then-hibernate.target]: OK, adjusted

**Comment by hadess** [comment]
I just remembered about `HybridSleep`. I didn't look at the docs fully, but can you add somewhere the main differences between the 2 and why one might want to prefer one over the other? Is there any reason why front-ends should just show "Suspend" and automatically use "Suspend Then Hibernate" if Hibernation is supported?

**Comment by superm1** [comment]
@hadess I think that https://github.com/systemd/systemd/blob/master/man/systemd-sleep.conf.xml#L93 explains what each of them does and I think does a good job contrasting them.

**Comment by superm1** [comment]
Regarding your question of what a frontend should show, that sounds reasonable enough to me.  Something worth noting though is that if "suspend-then-hibernate" failed, the frontend would probably want to fall back to "suspend" in that scenario.

**Comment by hadess** [comment]
> @hadess I think that https://github.com/systemd/systemd/blob/master/man/systemd-sleep.conf.xml#L93 explains what each of them does and I think does a good job contrasting them.

So hybrid-sleep is s2idle + hibernation, at the same time. suspend-then-hibernate is firmware-backed suspend followed by hibernation after a timeout. Is that correct?

**Comment by superm1** [comment]
In some situations yes that's correct.  They can both happen with firmware implemented suspend.

* hybrid-sleep is either s2idle or s3 + hibernation at the same time
**Pros:**
-If your battery goes dead you don't lose data
**Cons:**
-Performance is much worse when you suspend.

* suspend-then-hibernate is s2idle or s3 followed by timeout into hibernate
**Pros:**
-If your battery goes dead over an extended period of time you don't lose data
-Great performance for suspending
**Cons:**
-You may have to press the button to wakeup the system if hibernated
-You may have a longer wakeup time

**Pull request merged**